cmd/run-function/util.go

```diff
@@ -0,0 +1,89 @@
+package run_function
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"strings"
+)
+
+// WriteOutput Write the resource to the output directory.
+// If the output directory is not specified, the resource is written to stdout.
+func WriteOutput(outDir string, content string) error {
+	r := strings.NewReader(content)
+	var outputs []kio.Writer
+	outDir, err := GetDestinationDir(outDir)
+	if err != nil {
+		return err
+	}
+	if err := os.MkdirAll(outDir, 0755); err != nil {
+		return err
+	}
+	outputs = []kio.Writer{kio.LocalPackageWriter{PackagePath: outDir}}
+	return kio.Pipeline{
+		Inputs:  []kio.Reader{&kio.ByteReader{Reader: r}},
+		Outputs: outputs}.Execute()
+}
+
+// GetDestinationDir returns the destination directory, if outDir is not defined it creates a temp dir.
+func GetDestinationDir(outDir string) (string, error) {
+	if outDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return "", fmt.Errorf("failed to get current directory: %v", err)
+		}
+		dir, err := ioutil.TempDir(cwd, "crane_output")
+		if err != nil {
+			return "", err
+		}
+		outDir = dir
+	}
+	return outDir, nil
+}
+
+// ValidateFunctionImageURL validates the function name.
+// According to Docker implementation
+// https://github.com/docker/distribution/blob/master/reference/reference.go. A valid
+// name definition is:
+//	name                            := [domain '/'] path-component ['/' path-component]*
+//	domain                          := domain-component ['.' domain-component]* [':' port-number]
+//	domain-component                := /([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])/
+//	port-number                     := /[0-9]+/
+//	path-component                  := alpha-numeric [separator alpha-numeric]*
+// 	alpha-numeric                   := /[a-z0-9]+/
+//	separator                       := /[_.]|__|[-]*/
+//https://github.com/GoogleContainerTools/kpt/blob/main/pkg/api/kptfile/v1/validation.go#L120-L150
```

Comment:
nit. use a permalink

Reply:
done!

Reply:
Okay, I didn't know this but thanks for adding it here.
Now, I've updated it correctly